packages/frontend/src/components/common/ResourceList/ResourceIcon.tsx

```diff
@@ -28,13 +28,9 @@ const ResourceIcon: FC<ResourceIconProps> = ({ resource, resourceType }) => {
                 </ResourceIconBox>
             );
         case 'chart':
-            switch ((resource as SpaceQuery).chartType) {
-                case ChartKind.LINE:
-                    return (
-                        <ResourceIconBox color={Colors.BLUE3}>
-                            <IconChartLine color={Colors.BLUE3} size={20} />
-                        </ResourceIconBox>
-                    );
+            const chartType = (resource as SpaceQuery).chartType;
```

Comment:
why is the type casting needed here?

Reply:
resource can be either Dashboard `(DashboardBasicDetails)` or Chart `(SpaceQuery)`. I do not use type narrowing here and explicitly cast based on switch case which checks resourceType.

Reply:
I can update if that is bothersome